SAM/Lambda: show Toolkit-owned README file for new apps (#1885)
aws/aws-toolkit-vscode: resources/markdown/samReadme.md

# Developing AWS SAM Applications with the ${PRODUCTNAME}

> 

Reviewer: One of the main questions we are trying to address, is "what in the world is SAM". We should explain this very simply, non-pedantically, in one brief sentence:

> A "SAM" (serverless application model) project is just a project that contains a template.yaml file which is understood by AWS tooling (such as `sam` CLI, and AWS Toolkit for your IDE).

Author: The idea was that the user has to have an idea of what SAM is since they had to install it (I guess it comes for free with Cloud9). I can play with this, at the very least it's probably worth mentioning Lambda.

Reviewer: One brief sentence can avoid all doubt. Also, users are coming to "Create SAM Lambda" where they just want to work on Lambda. That's one of the main use-cases we're trying to address. They don't necessarily grok SAM or serverless.

> This project contains source code and supporting files for a serverless application that you can locally run, debug, and deploy to AWS with the ${PRODUCTNAME}.

Reviewer: We need to be as un-verbose as possible.
```suggestion
This project contains source code for a serverless ("SAM") application that you can debug locally and deploy to AWS.
```

Author: This line is the one and only direct rip I took from the existing readme 🙃 

https://github.com/aws/aws-sam-cli-app-templates/blob/master/nodejs14.x/cookiecutter-aws-sam-hello-nodejs/%7B%7Bcookiecutter.project_name%7D%7D/README.md#-cookiecutterproject_name-

Reviewer: yeah. We will keep narrowing and refining the text here over time. We have that luxury, because we only have 1 file to focus on :)

> ## Writing and Debugging Serverless Applications

Reviewer 2: We should include some information somewhere about setting up your toolchain to be able to debug. Maybe link directly to `https://docs.aws.amazon.com/toolkit-for-vscode/latest/userguide/setup-toolchain.html` (could be in the resources section)?

Author: Good thought. The difficult part is sliding this into Cloud9 as well since there's no toolchain required for that.

Functions that haven't been added to the [`template.yaml`](./template.yaml) file can also be directly invoked and debugged by creating a launch configuration through the ${CODELENS} over the function declaration, for quick debugging prior to committing the function to the SAM template.

## Deploying Serverless Applications

You can deploy a serverless application by invoking the `AWS: Deploy SAM application` command through the Command Palette or by right-clicking the Lambda node in the ${COMPANYNAME} Explorer and entering the deployment region, a valid S3 bucket from the region, and the name of a CloudFormation stack to deploy to. You can monitor your deployment's progress through the **${COMPANYNAME} Toolkit\*\* Output Channel.

## Interacting With Deployed Serverless Applications

A successfully-deployed serverless application can be found in the ${COMPANYNAME} Explorer under region and CloudFormation node that the serverless application was deployed to.

> Lambda Functions tied to the serverless application can be invoked by right-clicking the Lambda node and selecting "Invoke on AWS".

Reviewer: Avoid "tied to" in documentation and comments. It is vague and colloquial.  (In programmatic contexts "mapped to" or "associated with" are conventional, with the former implying directionality and the latter implying loose coupling.)

In this case, "tied to" is misleading since any function can be invoked remotely regardless of its association with the current project. So it's unnecessary detail, which we would strive deeply to avoid in this readme.

```suggestion
In the ${COMPANYNAME} Explorer, you can invoke *remote* AWS Lambda Functions by right-clicking the Lambda node and selecting "Invoke on AWS".
```
